CMakeLists.txt

```diff
@@ -34,6 +34,59 @@ endif()
 
 find_ros1_package(std_msgs REQUIRED)
 
+# ROS1 message packages need to be first in the PYTHONPATH
+# So we remove every ROS2 path from the python path for now
+set(__AMENT_PREFIX_PATH $ENV{AMENT_PREFIX_PATH})
+set(__ORIG_PYTHONPATH $ENV{PYTHONPATH})
+set(__PYTHONPATH "${__ORIG_PYTHONPATH}")
+string(REPLACE ":" ";" __AMENT_PREFIX_PATH ${__AMENT_PREFIX_PATH})
+string(REPLACE ":" ";" __PYTHONPATH ${__PYTHONPATH})
+foreach(ament_path ${__AMENT_PREFIX_PATH})
+  foreach(py_path ${__PYTHONPATH})
+    if(py_path MATCHES "^${ament_path}")
+      list(REMOVE_ITEM __PYTHONPATH ${py_path})
+    endif()
+  endforeach()
+endforeach()
+set(__PYTHONPATH_STRING "")
+foreach(py_path ${__PYTHONPATH})
+  string(CONCAT __PYTHONPATH_STRING "${__PYTHONPATH_STRING}" "${py_path}" ":")
+endforeach()
+
+set(ENV{PYTHONPATH} "${__PYTHONPATH_STRING}")
+
+# find all known ROS1 message/service packages
+execute_process(
+    COMMAND "rosmsg" "list"
+    OUTPUT_VARIABLE rosmsg_output
+    ERROR_VARIABLE rosmsg_error
+)
+execute_process(
+    COMMAND "rossrv" "list"
+    OUTPUT_VARIABLE rossrv_output
+    ERROR_VARIABLE rossrv_error
+)
+set(ENV{PYTHONPATH} ${__ORIG_PYTHONPATH})
```

Comment:
oh good, I didn't know cmake had a shell + platform agnostic way to do that, I'll look into it then